tests/test_cli.py

```diff
@@ -545,8 +546,18 @@ def test_refactor_product_should_not_warning_if_comment(tmp_empty):
 ])
 def test_test_notebook_runs(tmp_empty, code, output):
     nb_ = jupytext.reads(code, fmt='py:light')
-    for filename in ['nb.ipynb', 'nb.py']:
-        jupytext.write(nb_, filename)
-        runner = CliRunner()
-        result = runner.invoke(cli.test, filename)
-        assert output in result.output
+    filenames = ['nb.ipynb', 'nb.py']
+    output_paths = ["nb-output.ipynb", None]
+    for filename in filenames:
+        for output_path in output_paths:
+            shutil.rmtree(str(output_path), ignore_errors=True)
+            jupytext.write(nb_, filename)
+            runner = CliRunner()
+            if output_path:
+                expected_output_path = output_path
+                result = runner.invoke(cli.test, [filename, output_path])
+            else:
+                expected_output_path = 'nb-soorgeon-test.ipynb'
+                result = runner.invoke(cli.test, [filename])
+            assert output in result.output
+            assert Path(expected_output_path).exists()
```

Comment:
since we'll now be raising an exception, we have to test that the exit code is 1. see this https://click.palletsprojects.com/en/8.1.x/testing/#id4

Reply:
Added exit code tests